src/extractor.rs

```diff
@@ -1090,6 +1099,50 @@ mod tests {
         assert_eq!(*Path::<i8>::from_request(&req, &&PathConfig::default()).unwrap(), 32);
     }
 
+    #[test]
+    fn test_extract_path_decode() {
```

Comment:
Could you maybe add few test cases from the relevant issue like:
decoding `file%2F%2Fvar%2Flog%2Fsyslog`

Reply:
I added several cases from the issue